src/main/java/hudson/tasks/MailSender.java

```diff
@@ -177,26 +179,31 @@ protected MimeMessage getMail(AbstractBuild<?, ?> build, BuildListener listener)
         } catch (NoSuchMethodException x) {
             // non-deprecated subclass
         }
+
         if (build.getResult() == Result.FAILURE) {
             return createFailureMail(build, listener);
         }
-
         if (build.getResult() == Result.UNSTABLE) {
-            if (!dontNotifyEveryUnstableBuild)
+            if (!dontNotifyEveryUnstableBuild) {
                 return createUnstableMail(build, listener);
+            }
             Result prev = findPreviousBuildResult(build);
-            if (prev == Result.SUCCESS || prev == null)
+            if (prev == Result.SUCCESS || prev == null) {
                 return createUnstableMail(build, listener);
+            }
         }
 
         if (build.getResult() == Result.SUCCESS) {
             Result prev = findPreviousBuildResult(build);
-            if (prev == Result.FAILURE)
+            if (prev == Result.FAILURE) {
                 return createBackToNormalMail(build, Messages.MailSender_BackToNormal_Normal(), listener);
-            if (prev == Result.UNSTABLE)
+            }
+            if (prev == Result.UNSTABLE) {
                 return createBackToNormalMail(build, Messages.MailSender_BackToNormal_Stable(), listener);
+            }
         }
 
+        LOGGER.info("No mail will be sent out, as the current build's result is not set at all. Please make sure you set a proper result in case of pipeline/build scripts.");
```

Comment:
This message goes to a system log, but there is no build reference in it. "current build" reference is not enough, adding a full build name would be a good change.

I advice sending the message to the listener so that it goes to the build log

Reply:
Thanks for reviewing the requested change ... not sure what you mean by "listener", but I've adapted the message. Anything more?